src/model/Snapping.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Snapping.h"
+
+#include <cmath>
+namespace Snapping {
+[[nodiscard]] inline double roundToMultiple(double val, double multiple) { return val - std::remainder(val, multiple); }
+[[nodiscard]] inline double distance(Point const& a, Point const& b) { return std::hypot(b.x - a.x, b.y - a.y); }
+Point snapToGrid(Point const& pos, double gridSize, double tolerance) {
+    double abs_tolerance = (gridSize / sqrt(2)) * tolerance;
+    Point ret{roundToMultiple(pos.x, gridSize), roundToMultiple(pos.y, gridSize)};
```

Comment:
Since you are using the existing Point struct with the z coordinate/Pressure, you must copy the z value into the result. Otherwise you would throw away the pressure.

```suggestion
    Point ret{roundToMultiple(pos.x, gridSize), roundToMultiple(pos.y, gridSize), pos.z};
```